reframe/utility/sanity.py

```diff
@@ -488,7 +488,7 @@ def assert_reference(val, ref, lower_thres=None, upper_thres=None, msg=None):
 
     if upper_thres is not None:
         try:
-            evaluate(assert_bounded(upper_thres, 0, 1))
+            evaluate(assert_bounded(upper_thres, 0, None))
```

Comment:
Extending only the upper bound won't work correctly for negative reference values. It should be `None` both ways.